plugins/catalog/src/js/repositories/components/AddRepositoryFormModal.js

```diff
@@ -62,18 +63,19 @@ class AddRepositoryFormModal extends React.Component {
         min: "0",
         max: `${numberOfRepositories}`,
         step: "1",
+        disabled: noRepositories,
         validationErrorText: i18n._(
           t`Must be a positive integer between 0 and ${numberOfRepositories} representing its priority. 0 is the highest and ${numberOfRepositories} denotes the lowest priority.`
```

Comment:
This message is wrong https://docs.mesosphere.com/1.10/deploying-services/package-api/#/default/POST_package_repository_add `0` would mean highest prio, while _not providing any value_ would mean lowest prio.

Believe we should just validate that the number is `>= 0` and be done with it. No conditional validation whatsoever.
Thanks for keeping this open!

Reply:
But the priority field must be filled before we can add a repository, right?

And for validation you mean we should change
```
        validation(value) {
          return (
            ValidatorUtil.isDefined(value) &&
            ValidatorUtil.isNumberInRange(value, { max: numberOfRepositories })
          );
```
to just validating that `value` is `>=0`?

Reply:
> But the priority field must be filled before we can add a repository, right?

not if you want to add it with the lowest priority. According to documentation the field is optional @takirala to correct me please 🙇

Reply:
> to just validating that value is >=0?

Yes, I believe the validation should be changed completely not adjusted. So that we only validate that the number is `>= 0` or the field is empty altogether.

Reply:
Maybe it should be, but it isn't.
![peek 2019-01-28 15-59](https://user-images.githubusercontent.com/40791275/51840831-d23cdc00-2315-11e9-9afc-b3547f716a2e.gif)